gammapy/datasets/core.py

```diff
@@ -331,6 +331,12 @@ def to_spectrum_datasets(self, region):
 
         return datasets
 
+    def to_asimov_datasets(self):
+        """Create Asimov datasets from the current models."""
+        return Datasets(
+            [d._to_asimov_dataset(name=d.name, models=self.models) for d in self]
```

Comment:
why are you calling the hidden function here?

Reply:
removed